### PR TITLE
utils.flop_counter: add FLOP estimator and test for torch.fft.rfft

### PR DESCRIPTION
**Pull request by chall37**
### Summary
Adds FLOP estimation support for `torch.fft.rfft` by registering a formula
for `aten._fft_r2c` in `torch.utils.flop_counter`. The estimator uses the
standard 2.5 * batch * n * log2(n) model and includes a dedicated unit test
verifying correct decomposition and expected counts.

### Test Plan
Added `TestFlopCounterRFFT` in `test/test_flop_counter.py`.
All existing tests pass.

### Release Notes
- **torch.utils.flop_counter**: Added FLOP estimation support for `torch.fft.rfft`.

### Labels
`release notes: utils`
`topic: feature`
Fixes #165425 

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @aditew01

### CODE REVIEW

**Comment by msaroufim** [comment]
Is there a referencence on the math you could potentially link on how you got the flop estimation? I'm rusty on my FFT math

**Comment by chall37** [comment]
Of course, I should have justified my selection.  FFTs are O(N log N) [by definition](https://www.ams.org/journals/mcom/1965-19-090/S0025-5718-1965-0178586-1/S0025-5718-1965-0178586-1.pdf).  It's my understanding that the constants of 5 (for FFTs) and 2.5 (for RFFTs) from the radix-2 FFT are still used as the de facto standard, despite improvements in the algorithm over the years.

Happy to drop in a different estimate though, and/or wire up a (complex) FFT FLOP "count" as well.

> To report FFT performance, we plot the "mflops" of each FFT, which is a scaled version of the speed, defined by:
> mflops = 5 N log2(N) / (time for one FFT in microseconds) for complex transforms, and
> mflops = 2.5 N log2(N) / (time for one FFT in microseconds) for real transforms]  https://www.fftw.org/speed/method.html

> https://hpkfft.com/pdf/hpkfft-OnFlopFlapCount.pdf

> [Danielson-Lanczos Lemma](https://mice.cs.columbia.edu/getTechreport.php?techreportID=1152&format=pdf)

> The discrete Fourier transform can, in fact, be computed in O(N log2 N) operations with an algorithm called the fast
> Fourier transform, or FFT.  https://faculty.washington.edu/seattle/brain-physics/FFT/numerical-recipes.pdf
